cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -863,7 +863,14 @@ protected Affinity getMergedAffinity() {
                 .withVolumeMounts(getVolumeMounts())
                 .withPorts(getContainerPortList())
                 .withLivenessProbe(createTcpSocketProbe(REPLICATION_PORT, livenessInitialDelay, livenessTimeout))
```

Comment:
I think you still have to fix the liveness probe. IMHO the problem is that while recovering the files, the listeners are not yet open. So the liveness probe will be failing and killing the pod.

Reply:
That's an excellent point. Thank you! I have a cunning plan which might manage to avoid needing to spin up JVM processes.